draft-ietf-ice-pac.xml

```diff
@@ -66,31 +66,38 @@
         for Network Address Translator (NAT) traversal for UDP-based communication.
       </t>
       <t>
-        Typically, when using ICE, endpoints will exchange candidate addresses,
-        form a list of candidate pairs, and then test each candidate pair to see
-        if connectivity can be established. If the test for a given pair fails,
-        it is marked accordingly, and if all pairs have failed, the overall
-        ICE process is considered to have failed.
+        When using ICE, endpoints will exchange candidiates, form one or more
+        lists (checklists) of candidate pairs, and then test the pairs to
+        determine whether connectivity can be established. There is a checklist
+        for each data stream associated with the session. If connectivity for
+        a data stream cannot be established, the state of the checklist associated
+        with the data stream is set to Failed. Once the state of all checklists
+        have been set to Failed, the ICE agent sets the state of the ICE session
+        to Failed. Based on local policy, the ICE agent can also set the state of 
+        the ICE session to Failed even if the state of only some of the checklists
+        have been set to Failed.
       </t>
       <t>
-        During the process of connectivity checks, additional candidates may
-        be created as a result of successful inbound checks from the remote
-        peer. Such candidates are referred to as peer-reflexive candidates,
-        and once discovered, will be used to form new candidate pairs which will
-        be tested like any other. However, there is an inherent race condition
-        here; if, before learning about any peer-reflexive candidates, an
-        endpoint runs out of candidate pairs to check, either because it has
-        none, or it considers them all to have failed, it will prematurely
-        declare failure and terminate ICE processing. This race condition can
-        occur in many common situations.
+        While candidate pairs are tested, additional candidates, referred to as
+        peer-reflexive candidates, might be created as a result of successful
+        inbound ICE connectitivy checks received from the remote peer. Such
+        candidates will be used to form new candidate pairs that will be tested
+        like any other candidate pair.
```

Comment:
Could we have a separate PR for reformatting things that don't change the meaning so we can focus on just the diff that changes the meaning?

Reply:
Fair enough.

Reply:
Fair enough.